src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -567,4 +571,64 @@ public virtual void PrintMessageNormalized(TextWriter writer, string message, bo
         else if (!removeLastNewLine)
             writer.WriteLine();
     }
+
+    /// <summary>
+    /// Adds a new runtime option. Throws if option already exists.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="name"></param>
+    /// <param name="value"></param>
+    public void AddOption<T>(string name, T value)
+    {
+        Options.Add(name, value);
+    }
+
+    /// <summary>
+    /// Trys to add a new runtime option.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="name"></param>
+    /// <param name="value"></param>
+    /// <returns>True if successful. False otherwise.</returns>
+    public bool TryAddOption<T>(string name, T value)
+    {
+        if (Options.ContainsKey(name))
+            return false;
+        AddOption<T>(name, value);
+        return true;
+    }
+
+    public void AddOrOverwriteOption<T>(string name, T value)
+    {
+        if (!Options.ContainsKey(name))
+            AddOption<T>(name, value);
+        else
+            Options[name] = value;
+    }
+
+    public T GetOption<T>(string name)
+    {
+        return (T)Options[name];
+    }
+
+    public T GetOptionOrDefault<T>(string name)
+    {
+        if (!Options.ContainsKey(name))
+            AddOption<T>(name, default);
```

Comment:
>default [](http://example.com/codeflow?start=31&length=7)

This will need you define the value as `object?`

Reply:
It didn't make me define it that way? But since I'm going to make that change anyways its ok.

Reply:
Since we don't have nullable context enabled here skipping this for now.